java/opendcs/src/test/java/decodes/cwms/resevapcalc/ResWaterTemperatureComputeTest.java

```diff
@@ -0,0 +1,67 @@
+package decodes.cwms.resevapcalc;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+/**
+ * Testing vertically averaged density within the vertical temp profile computation using average density.
+ * Volumetric average is included to compare results.
+ */
+class ResWaterTemperatureComputeTest
+{
+    private static final double[] RHOW_VALUES = {997.0, 998.5, 1000.0, 1001.0, 1003.0}; // Example densities in kg/m^3
+    private static final double[] DEPTH_VALUES = {0.5, 1.0, 2.0, 3.0, 4.0}; // Example depths in meters
+    private static final double[] VOLUME_VALUES = {1000.0, 2500.0, 6000.0, 12000.0, 20000.0};
+    private static final double[] RHOW_DEPTH_PRODUCTS = new double[RHOW_VALUES.length];
+    private static final double[] RHOW_VOLUME_PRODUCTS = new double[RHOW_VALUES.length];
+
+    static
+    {
+        for (int i = 0; i < RHOW_VALUES.length; i++)
+        {
+            RHOW_DEPTH_PRODUCTS[i] = RHOW_VALUES[i] * DEPTH_VALUES[i];
+            RHOW_VOLUME_PRODUCTS[i] = RHOW_VALUES[i] * VOLUME_VALUES[i];
+        }
+    }
+
+
+    @MethodSource("getLayerIndexValues")
```

Comment:
The inputs to a parameterized test should be the individual values, not a single index to multiple arrays.